From b997b885f468bb2ef20a70133bc194a27fa27a8e Mon Sep 17 00:00:00 2001
From: Oswald Buddenhagen <oswald.buddenhagen@digia.com>
Date: Mon, 29 Oct 2012 10:30:33 +0100
Subject: [PATCH] don't set QT_BUILD_DIR as a side effect

we would be setting QT_BUILD_DIR to $$[QT_INSTALL_DATA] as a fallback,
which would make qtPrepareTool() believe we were doing a qt build
despite actually trying to build something against an installed qt. this
would become apparent only when using a non-default directory layout, as
is the case in our mac packages.

Change-Id: I4f27e9427222ad3b9bb83fcc625bab19e00db9be
Reviewed-by: Miikka Heikkinen <miikka.heikkinen@digia.com>
Reviewed-by: Teemu Katajisto <teemu.katajisto@digia.com>
Reviewed-by: Joerg Bornemann <joerg.bornemann@digia.com>
---
 mkspecs/features/device_config.prf |    9 +++++----
 mkspecs/features/qt_config.prf     |    8 +++++---
 2 files changed, 10 insertions(+), 7 deletions(-)

diff --git a/mkspecs/features/device_config.prf b/mkspecs/features/device_config.prf
index aa0aa50..31a9580 100644
--- a/mkspecs/features/device_config.prf
+++ b/mkspecs/features/device_config.prf
@@ -1,8 +1,9 @@
 # Load generated qdevice.pri
-isEmpty(QT_BUILD_TREE):exists($$_QMAKE_CACHE_): QT_BUILD_TREE = $$fromfile($$_QMAKE_CACHE_, QT_BUILD_TREE)
-isEmpty(QT_BUILD_TREE): QT_BUILD_TREE = $$[QT_INSTALL_DATA]
-
-DEVICE_PRI = $$QT_BUILD_TREE/mkspecs/qdevice.pri
+qdd = $$QT_BUILD_TREE
+isEmpty(qdd):exists($$_QMAKE_CACHE_): qdd = $$fromfile($$_QMAKE_CACHE_, QT_BUILD_TREE)
+isEmpty(qdd): qdd = $$[QT_INSTALL_DATA]
+DEVICE_PRI = $$qdd/mkspecs/qdevice.pri
+unset(qdd)
 
 exists($$DEVICE_PRI):include($$DEVICE_PRI)
 unset(DEVICE_PRI)
diff --git a/mkspecs/features/qt_config.prf b/mkspecs/features/qt_config.prf
index 4575f4a..fc14cdd 100644
--- a/mkspecs/features/qt_config.prf
+++ b/mkspecs/features/qt_config.prf
@@ -3,9 +3,11 @@
 
 exists($$_QMAKE_CACHE_):QMAKE_QT_CONFIG = $$fromfile($$_QMAKE_CACHE_, QMAKE_QT_CONFIG)
 isEmpty(QMAKE_QT_CONFIG)|!exists($$QMAKE_QT_CONFIG) {
-   isEmpty(QT_BUILD_TREE):exists($$_QMAKE_CACHE_): QT_BUILD_TREE = $$fromfile($$_QMAKE_CACHE_, QT_BUILD_TREE)
-   isEmpty(QT_BUILD_TREE): QT_BUILD_TREE = $$[QT_INSTALL_DATA]
-   QMAKE_QT_CONFIG = $$QT_BUILD_TREE/mkspecs/qconfig.pri
+   qdd = $$QT_BUILD_TREE
+   isEmpty(qdd):exists($$_QMAKE_CACHE_): qdd = $$fromfile($$_QMAKE_CACHE_, QT_BUILD_TREE)
+   isEmpty(qdd): qdd = $$[QT_INSTALL_DATA]
+   QMAKE_QT_CONFIG = $$qdd/mkspecs/qconfig.pri
+   unset(qdd)
 }
 !include($$QMAKE_QT_CONFIG, "", true) {
    debug(1, "Cannot load qconfig.pri!")
-- 
1.7.1
